File: Niveau 3/10 – Bases/6 - Changement de base.py
def convertToB10(n, base):
    base10 = 0
    n = n[::-1]
    for i in range(len(n)):
        base10 += n[i] * (base ** i)
    return base10
def convertToAnyBase(n, base):
    arr = []
    while n > 0:
        arr.append(n % base)
        n = n // base
    if len(arr) == 0:
        arr.append(0)
    return arr[::-1]

def main():
    baseD, baseE, length = map(int, input().split())
    numbers = list(map(int, input().split()))
    b10 = convertToB10(numbers, baseD)
    n = convertToAnyBase(b10, baseE)
    for i in range(len(n)):
        print(n[i], end=" ")
main()

File: Niveau 3/10 – Bases/6 - Changement de base.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

int convertToB10(vector<int> nb, int base)
{
  int base10 = 0;
  for (int i = 0; i < (int)nb.size(); i++)
  {
    base10 += nb[i] * (pow(base, (int)nb.size() - i - 1));
  }
  return base10;
}

void writeInAnyBase(int nb, int base)
{
  vector<int> v;
  while (nb > 0)
  {
    v.push_back(nb % base);
    nb /= base;
  }
  if (v.size() == 0)
  {
    cout << "0";
  }
  else
  {
    for (int i = (int)v.size() - 1; i >= 0; i--)
    {
      cout << v[i] << ' ';
    }
  }
}

int main()
{
  ios::sync_with_stdio(false);
  int baseD, baseE, l;
  cin >> baseD >> baseE >> l;
  vector<int> nombres;
  for (int i = 0; i < l; i++)
  {
    int n;
    cin >> n;
    nombres.push_back(n);
  }
  int base10 = convertToB10(nombres, baseD);
  writeInAnyBase(base10, baseE);
}